packages/frontend-api/src/Model/Mutation/Complaint/CreateComplaintMutation.php

```diff
@@ -66,14 +70,14 @@ public function createComplaintMutation(Argument $argument, InputValidator $vali
 
         $order = $this->orderApiFacade->getByUuid($input['orderUuid']);
         $customerUser = $this->currentCustomerUser->findCurrentCustomerUser();
+        $orderCustomerUser = $order->getCustomerUser();
 
-        if ($order->getCustomerUser() !== $customerUser) {
+        if ($orderCustomerUser !== $customerUser && !$this->security->isGranted(CustomerUserRole::ROLE_API_ALL)) {
```

Comment:
I have a bad feeling about this - with the `ROLE_API_ALL` role, I would be able to create a complaint for any order? Could you please add tests for this behavior?

Reply:
I'm not sure how it was missed, but I remember adding the customer check here yesterday. I've now fixed it.

[fixup! complaints b2b customers support](https://github.com/shopsys/shopsys/pull/3295/commits/585c5b09e615c10c4351fc201d7abf686fa2e022)

Reply:
I checked with the PO, and your approach is preferred. Updated. [fixup! complaints b2b customers support](https://github.com/shopsys/shopsys/pull/3295/commits/585c5b09e615c10c4351fc201d7abf686fa2e022)